var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -139,21 +141,14 @@ def install(self, spec, prefix):
                 '--disable-hl',
             ])
 
-        configure(
-            "--prefix=%s" % prefix,
-            "--with-zlib=%s" % spec['zlib'].prefix,
-            *extra_args)
-        make()
+        return ["--with-zlib=%s" % spec['zlib'].prefix] + extra_args
 
-        if self.run_tests:
-            make("check")
-
-        make("install")
-        self.check_install(spec)
-
-    def check_install(self, spec):
+    @AutotoolsPackage.sanity_check('install')
+    @AutotoolsPackage.on_package_attributes(run_tests=True)
+    def check_install(self):
```

Comment:
I mentioned this before, but I'm still not very happy with having to add:
```python
@AutotoolsPackage.sanity_check('install')
@AutotoolsPackage.on_package_attributes(run_tests=True)
```
before every single `check_install()`. It would be nice if this could be handled in the AutotoolsPackage class so that people don't have to copy-n-paste this every time they want to add a test.